flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -155,43 +156,49 @@ public void deregisterApplication(
             TaskExecutorProcessSpec taskExecutorProcessSpec) {
         final KubernetesTaskManagerParameters parameters =
                 createKubernetesTaskManagerParameters(taskExecutorProcessSpec);
-        final KubernetesPod taskManagerPod =
-                KubernetesTaskManagerFactory.buildTaskManagerKubernetesPod(
-                        taskManagerPodTemplate, parameters);
-        final String podName = taskManagerPod.getName();
         final CompletableFuture<KubernetesWorkerNode> requestResourceFuture =
                 new CompletableFuture<>();
 
-        requestResourceFutures.put(podName, requestResourceFuture);
-
-        log.info(
-                "Creating new TaskManager pod with name {} and resource <{},{}>.",
-                podName,
-                parameters.getTaskManagerMemoryMB(),
-                parameters.getTaskManagerCPU());
-
-        final CompletableFuture<Void> createPodFuture =
-                flinkKubeClient.createTaskManagerPod(taskManagerPod);
-
-        FutureUtils.assertNoException(
-                createPodFuture.handleAsync(
-                        (ignore, exception) -> {
-                            if (exception != null) {
-                                log.warn(
-                                        "Could not create pod {}, exception: {}",
-                                        podName,
-                                        exception);
-                                CompletableFuture<KubernetesWorkerNode> future =
-                                        requestResourceFutures.remove(taskManagerPod.getName());
-                                if (future != null) {
-                                    future.completeExceptionally(exception);
+        try {
+            final KubernetesTaskManagerSpecification taskManagerSpec =
+                    KubernetesTaskManagerFactory.buildTaskManagerKubernetesPod(
```

Comment:
Why are we now wrapping everything in the try-catch block instead of only the `buildTaskManagerKubernetesPod` part?